### PR TITLE
Refactor e2e-tester.go for event-bus

### PR DESCRIPTION
**Pull request by anishj0shi**
w.ref.t. #4265 

1) Structify global variables and use their instances in relevant functions
2) Table Driven Checks within checkSubscriberReceivedEventHeaders in lieu of Cascading Conditional statements

### CODE REVIEW

File: tests/event-bus/e2e-tester/e2e-tester.go [hunk 1]
```diff
@@ -412,30 +424,30 @@ func checkSubscriberReceivedEventHeaders() error {
 			lowerResponseHeaders[strings.ToLower(k)] = resp[k]
 		}
 
-		if lowerResponseHeaders[sourceHeader][0] != srcID {
-			return fmt.Errorf("wrong response: %s, want: %s", lowerResponseHeaders[sourceHeader][0], srcID)
+		var testDataSets = []struct {
```
Comment (nachtmaar) [inline tests/event-bus/e2e-tester/e2e-tester.go]: Why do you need a new struct? Isn't is just key/value pairs? So a map could be a simply alternative.

Comment (anishj0shi) [reply tests/event-bus/e2e-tester/e2e-tester.go]: i was trying to follow this paradigm: https://github.com/golang/go/wiki/TableDrivenTests 

by having it in a struct i get compile time safety during iteration/

File: tests/event-bus/e2e-tester/e2e-tester.go [hunk 2]
```diff
@@ -220,53 +220,65 @@ func main() {
...
+	subscriber.subscriberResultsEndpointURL = "http://" + util.SubscriberName + "." + subscriber.subscriberNamespace + ":9000/v1/results"
+	subscriber.subscriberStatusEndpointURL = "http://" + util.SubscriberName + "." + subscriber.subscriberNamespace + ":9000/v1/status"
+	subscriber.subscriberShutdownEndpointURL = "http://" + util.SubscriberName + "." + subscriber.subscriberNamespace + ":9000/shutdown"
+	subscriber.subscriber3EventEndpointURL = "http://" + util.SubscriberName + "." + subscriber.subscriberNamespace + ":9000/v3/events"
+	subscriber.subscriber3ResultsEndpointURL = "http://" + util.SubscriberName + "." + subscriber.subscriberNamespace + ":9000/v3/results"
+	subscriber.subscriber3StatusEndpointURL = "http://" + util.SubscriberName + "." + subscriber.subscriberNamespace + ":9000/v3/status"
 }
 
-func shutdown(code int) {
+func shutdown(code int, subscriber *subscriberDetails) {
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/event-bus/e2e-tester/e2e-tester.go]: makes sense, let me try that

File: tests/event-bus/e2e-tester/e2e-tester.go [hunk 3]
```diff
@@ -354,11 +366,11 @@ func publishHeadersTestEvent(publishEventURL string) (*api.Response, error) {
 	return respObj, err
 }
 
-func checkSubscriberReceivedEvent() error {
+func checkSubscriberReceivedEvent(subscriber *subscriberDetails) error {
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/event-bus/e2e-tester/e2e-tester.go]: makes sense, let me try that

File: tests/event-bus/e2e-tester/e2e-tester.go [hunk 4]
```diff
@@ -54,47 +54,47 @@ const (
...
+
+//Unexportable struct, encapsulates publisher details
+type publisherDetails struct {
+	publishEventEndpointURL  string
+	publishStatusEndpointURL string
+}
+
 func main() {
 	flags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	var subscriber subscriberDetails
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/event-bus/e2e-tester/e2e-tester.go]: on it!

File: tests/event-bus/e2e-tester/e2e-tester.go [hunk 5]
```diff
@@ -54,47 +54,47 @@ const (
...
+	var publisher publisherDetails
 
-	publishEventEndpointURL = flags.String("publish-event-uri", "http://event-bus-publish:8080/v1/events", "publish service events endpoint `URL`")
-	publishStatusEndpointURL = flags.String("publish-status-uri", "http://event-bus-publish:8080/v1/status/ready", "publish service status endpoint `URL`")
-	subscriberImage := flags.String("subscriber-image", "", "subscriber Docker `image` name")
-	subscriberNamespace = flags.String("subscriber-ns", "default", "k8s `namespace` in which subscriber test app is running")
-	subscriberEventEndpointURL = flags.String("subscriber-events-uri",
```
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/event-bus/e2e-tester/e2e-tester.go]: almost all the flags are derived via a host and a namespace param, we can derive this at runtime, less flags, I'll add a host flag which can be used as an optional parameter for deriving all the urls.

[the remark this answers is not in the document]
Comment (nachtmaar) [reply tests/event-bus/e2e-tester/e2e-tester.go]: Makes sense now :)

File: tests/event-bus/e2e-tester/e2e-tester.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/event-bus/e2e-tester/e2e-tester.go]: done

File: tests/event-bus/e2e-tester/e2e-tester.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/event-bus/e2e-tester/e2e-tester.go]: i evaluated this, for this particular method, it didn't make sense to make it part of subscriberDetails, for other methods i did those changes

File: tests/event-bus/e2e-tester/e2e-tester.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (anishj0shi) [reply tests/event-bus/e2e-tester/e2e-tester.go]: done

**Pull request merged**